toolchain/check/return.cpp

```diff
@@ -20,9 +20,7 @@ static auto GetCurrentFunction(Context& context) -> SemIR::Function& {
   return context.functions().Get(function_id);
 }
```

Comment:
Would it make sense to move this to `check/function.cpp` now that it's being used more widely? Oh, or, perhaps we should rename it to something like `GetCurrentFunctionForReturn` given that the new usage is only in `returned` handling, and what we really mean here is "get me the function that a `return` here would return from" not some other notion of current / enclosing function.